hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceShipper.java

```diff
@@ -290,7 +290,22 @@ private boolean updateLogPosition(WALEntryBatch batch) {
   public void startup(UncaughtExceptionHandler handler) {
     String name = Thread.currentThread().getName();
     Threads.setDaemonThreadRunning(this,
-      name + ".replicationSource.shipper" + walGroupId + "," + source.getQueueId(), handler);
+      name + ".replicationSource.shipper" + walGroupId + "," + source.getQueueId(),
+      (t,e) -> {
```

Comment:
OK, the code is almost the same... Then I think we could move the logic into uncaughtException method? If abortOnError is true, we about, otherwise we will try to refresh the source.

Reply:
Moved it to _uncaughtException_. This is true when handling potential errors on both the reader and shipper threads, but _uncaughtException_ is also used when handling issues from _ReplicationSource.initialize_ method, which might blow before reader/shipper is even started. So needed to add extra check in _uncaughtException_ to decide when to invoke _refreshSources_.